include/fi_mem.h

```diff
@@ -61,26 +61,39 @@ static inline void *mem_dup(const void *src, size_t size)
 	return dest;
 }
 
+struct util_buf_pool;
+typedef void (*util_buf_region_alloc_hndlr) (struct util_buf_pool *pool,
```

Comment:
I think this needs to return an int.  On failure, the allocated region should be freed, with the failure reported back through the call stack.